doc/transform.ipynb

```diff
@@ -0,0 +1,7402 @@
+       "       1.1634438 , 1.71898465, 1.04764537, 0.87075606, 1.27740045])</pre></li><li class='xr-var-item'><div class='xr-var-name'><span>theta</span></div><div class='xr-var-dims'>(z)</div><div class='xr-var-dtype'>float64</div><div class='xr-var-preview xr-preview'>0.6931 1.099 1.386 ... 2.303 2.398</div><input id='attrs-adb7ed65-9829-4116-983b-3a950e4d7742' class='xr-var-attrs-in' type='checkbox' disabled><label for='attrs-adb7ed65-9829-4116-983b-3a950e4d7742' title='Show/Hide attributes'><svg class='icon xr-icon-file-text2'><use xlink:href='#icon-file-text2'></use></svg></label><input id='data-346359c2-3401-44e5-8c73-9b863f2460c3' class='xr-var-data-in' type='checkbox'><label for='data-346359c2-3401-44e5-8c73-9b863f2460c3' title='Show/Hide data repr'><svg class='icon xr-icon-database'><use xlink:href='#icon-database'></use></svg></label><div class='xr-var-attrs'><dl class='xr-attrs'></dl></div><pre class='xr-var-data'>array([0.69314718, 1.09861229, 1.38629436, 1.60943791, 1.79175947,\n",
+       "       1.94591015, 2.07944154, 2.19722458, 2.30258509, 2.39789527])</pre></li></ul></div></li><li class='xr-section-item'><input id='section-1770c99e-ee52-4fb1-8331-fa688e4948ed' class='xr-section-summary-in' type='checkbox' disabled ><label for='section-1770c99e-ee52-4fb1-8331-fa688e4948ed' class='xr-section-summary'  title='Expand/collapse section'>Attributes: <span>(0)</span></label><div class='xr-section-inline-details'></div><div class='xr-section-details'><dl class='xr-attrs'></dl></div></li></ul></div></div>"
+      ],
+      "text/plain": [
+       "<xarray.Dataset>\n",
+       "Dimensions:  (z: 10, zc: 11)\n",
+       "Coordinates:\n",
+       "  * z        (z) int64 2 3 4 5 6 7 8 9 10 11\n",
+       "  * zc       (zc) float64 1.5 2.5 3.5 4.5 5.5 6.5 7.5 8.5 9.5 10.5 11.5\n",
+       "Data variables:\n",
+       "    phi      (z) float64 1.691 1.515 1.761 1.251 ... 1.719 1.048 0.8708 1.277\n",
+       "    theta    (z) float64 0.6931 1.099 1.386 1.609 ... 2.079 2.197 2.303 2.398"
+      ]
+     },
+     "execution_count": 8,
+     "metadata": {},
+     "output_type": "execute_result"
+    }
+   ],
+   "source": [
+    "# define the cell bounds in depth\n",
+    "zc = np.arange(1,12)+0.5\n",
+    "\n",
+    "# add them to the existing dataset\n",
+    "ds = ds.assign_coords({'zc': zc})\n",
+    "ds"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 9,
+   "metadata": {},
+   "outputs": [
+    {
+     "data": {
+      "text/plain": [
+       "<xgcm.Grid>\n",
+       "Z Axis (not periodic, boundary=None):\n",
+       "  * center   z --> outer\n",
+       "  * outer    zc --> center"
+      ]
+     },
+     "execution_count": 9,
+     "metadata": {},
+     "output_type": "execute_result"
+    }
+   ],
+   "source": [
+    "# Recreate the grid object with a staggered `center`/`outer` coordinate layout\n",
```

Comment:
Hmmm... this seems like a nice feature to add.

`Grid.add_coord({"Z": {"outer": "zc"}})  # inplace?`